packages/ts-moose-lib/src/consumption-apis/runner.ts

```diff
@@ -132,78 +152,122 @@ const apiHandler =
         {},
       );
 
-      let userFuncModule = modulesCache.get(pathName);
+      let cacheKey = endpointName;
+      if (isDmv2 && requestVersion) {
+        cacheKey = `${endpointName}_v${requestVersion}`;
+      }
+
+      let userFuncModule = modulesCache.get(cacheKey);
       if (userFuncModule === undefined) {
         if (isDmv2) {
-          const egressApis = await getEgressApis();
-          userFuncModule = egressApis.get(fileName.replace(/^\/+/, ""));
-          modulesCache.set(pathName, userFuncModule);
+          const egressApis = await getEgressApiInstances();
+
+          if (requestVersion) {
+            const versionedApiKey = `v${requestVersion}/${endpointName}`;
+            userFuncModule = egressApis.get(versionedApiKey);
+          } else {
+            userFuncModule = egressApis.get(endpointName);
+
+            if (!userFuncModule) {
+              const availableVersions: string[] = [];
+              for (const apiKey of egressApis.keys()) {
```

Comment:
We might not want to call this `apiKey`. It sounds like something for auth but it looks like just the api with possible version.